azurelinuxagent/common/cgroups.py

```diff
@@ -476,6 +485,22 @@ def add(self, pid):
             tasks_file = self._get_cgroup_file(hierarchy, 'cgroup.procs')
             fileutil.append_file(tasks_file, "{0}\n".format(pid))
 
+    def get_cpu_limits(self):
+        # default values
+        cpu_limit = DEFAULT_CPU_LIMIT_AGENT if AGENT_NAME.lower() in self.name.lower() else DEFAULT_CPU_LIMIT_EXT
```

Comment:
I don't see a mechanism to allow the user (or extension owner) to override the default limits for a specific extension. Suppose an extension owner determines that her extension (e.g. "Backup") regularly exceeds the default CPU limit, but decides that the reason for using the extension justifies that. How can they request the agent raise the limit for their extension?

Reply:
@jasonzio The current PR is only addressing the generating events when CPU/Memory thresholds are reached/crossed. The `get_[cpu,memory]_limits()` functions can later be used to read from a config file to read extension level memory and cpu limits.

We are currently planning to introduce the limits as a part of the extension_manifest, which would be honored by the agent, with a fallback value.